### PR TITLE
Build rules for producing and verifying ledger dumps

### PR DESCRIPTION
**Pull request by dajmaki**
This PR add the following bazel rules:
- client_server_build: Like client_server_test, but for producing output from running a client against a server
- integrity_test: For running the kvutils integrity check as a bazel test

And additionally this PR adds bazel targets to produce a ledger dump (using client_server_build)
from running the test tool against the reference server, and test target for verifying the ledger dump.

As a next step we will publish the ledger dumps as part of releases, once we have a place to publish
them to.

### Pull Request Checklist

- [ ] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/master/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Add a line to the [release notes](https://github.com/digital-asset/daml/blob/master/unreleased.rst), if appropriate
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

File: bazel_tools/client_server/tests/client.sh [hunk 1]
```diff
@@ -27,6 +27,12 @@ if [[ -z "$port" || -z "$inputFile" ]]; then
   exit 1
 fi
 
+
 exec 5<> /dev/tcp/localhost/$port
 cat $inputFile >&5
 cat <&5
+
+# Write the output file for testing client_server_build.
+if [ -n "${CS_OUT-}" ]; then
```
Comment (gerolf-da) [inline bazel_tools/client_server/tests/client.sh]: What's the significance of the `-` after the variable name?

**Comment by gerolf-da** [comment]
@aherrmann-da: could I ask you to review this PR as well from a bazel perspective?

File: bazel_tools/client_server/tests/client.sh (hunk 1 above)
Comment (aherrmann-da) [reply bazel_tools/client_server/tests/client.sh]: The script has `set -u` enabled, which means that `"${CS_OUT}"` would be an error if `CS_OUT` is not defined. With `"${CS_OUT-}"` (with the dash) the shell will substitute the empty string and not raise an error.

File: bazel_tools/client_server/client_server_build.bzl [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+            server_args = " ".join(ctx.attr.server_args),
+            server_files = " ".join([f.path for f in ctx.files.server_files]),
+            client_args = " ".join(ctx.attr.client_args),
+            client_files = " ".join([f.path for f in ctx.files.client_files]),
+        ),
+    )
+    return [
+        DefaultInfo(
+            files = depset([ctx.outputs.out]),
+            runfiles = ctx.runfiles(files = ctx.files.data, collect_data = True),
```
Comment (aherrmann-da) [inline bazel_tools/client_server/client_server_build.bzl]: Are files passed in via `data` really needed in `runfiles`? As I understand they are files required by the `client` or `server` during the execution of the `runner` command.
```suggestion
```

File: bazel_tools/client_server/tests/client.sh (hunk 1 above)
Comment (gerolf-da) [reply bazel_tools/client_server/tests/client.sh]: thank you, @aherrmann-da

File: bazel_tools/client_server/client_server_build.bzl (hunk 2 above)
Comment (dajmaki) [reply bazel_tools/client_server/client_server_build.bzl]: good point. wasn't needed.

**Pull request merged**